docs/src/how-to/install/infrastructure-configuration.md

```diff
@@ -677,3 +677,30 @@ cargohold:
 
 When this option is unspecified, wire-server defaults to path style addressing
 to ensure smooth transition for older deployments.
+
+## I have a team larger than 500 users
+
+By default, the maximum number of users in a team is set at 500.
+
+This can be changed in the Brig config, with this option:
+
+```yaml
+
+    optSettings:
+      setMaxTeamSize: 501
+
+```
+   
+```{note}
+
+   Note however that some messages need to be fanned out to all members of a team, and due to End to End Encryption (E2EE), this is very computationally expensive for large numbers of recipients.
+
+   Therefore, it is recommended to leave the maximum number of users at 500, and to instead create additional teams.
+
+   Very large teams could cause issues with dropped messages and with Legal Hold being unavailable.
```

Comment:
The backend event system (gundeck) is really the reason for these limitations. The Proteus e2ee protocol leads to computationally expensive operations for large groups, but it would still work. The backend is just not designed currently to handle large fanouts. We have it on the roadmap to scale gundeck.

```suggestion
If you create a team with more than 2000 members then clients won't receive certain team update events (e.g. new member joining) live via websocket anymore, but the most of the app will still function normally.

Irrespective of team size the every conversation can have at most 2000 members. This limit cannot be overridden.

Wire's backend currently only supports fanning out a single message to at most 2000 recipients, hence the limitations on conversation size. Increasing this limit is work in progress.
```